aten/src/ATen/native/mps/operations/Activation.mm

```diff
@@ -1575,7 +1575,7 @@ Tensor glu_backward_mps (const Tensor& grad_output,
       MPSStream* stream = getCurrentMPSStream();
 
       @autoreleasepool {
-        string key = "softplus_backward_out_mps:" + getTensorsStringKey({grad_output, self});
+        string key = "softplus_backward_out_mps:" + getTensorsStringKey({grad_output, self}) + ":" + to_string(threshold.to<double>());
```

Comment:
I understand that this fixes the issue with the varying threshold values. However, adding the constant value to the string key is inefficient because it recreates a new graph each time the threshold value changes. 
Instead, can you please create a `placeholder` for it like we do with the Beta value? For that, you'll need to:
1- replace this line:
`MPSGraphTensor* thresholdTensor = [mpsGraph constantWithScalar:threshold.to<double>()
                                                                       shape:@[@1]
                                                               dataType:getMPSDataType(self.scalar_type())];`
With something like this:
`MPSGraphTensor* thresholdTensor = mpsGraphScalarPlaceHolder(mpsGraph, ScalarType::Float);`
2- Add `thresholdTensor_` to `CachedGraph` struct, and add this as well:
`newCachedGraph->thresholdTensor_ = thresholdTensor;`
3- Pass the value of the threshold to the placeholder like this in the `feeds` array:
`cachedGraph->thresholdTensor_ : getMPSGraphTensorFromScalar(stream, threshold.to<double>())`

Thank you @qqaatw.

Reply:
done